apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -222,8 +224,12 @@ export function OAuth2Form() {
   )
 
   const onConsent = useCallback(() => {
-    sendGTagEvent('UserConsented')
-    if (oidcAuthorizationRequestFromSearchParams) {
+    if (
+      oidcAuthorizationRequestFromSearchParams &&
+      !hasCalledOnConsent.current
+    ) {
+      hasCalledOnConsent.current = true
+      sendGTagEvent('UserConsented')
       consentToRequest(oidcAuthorizationRequestFromSearchParams)
```

Comment:
Would it be possible to reproduce this in a test and verify that this is only called once?

Reply:
Good idea